Author: Ganagalla Joshitha
Project: Google Play Store Intelligence Dashboard
Role: Data Analytics Portfolio Project
Tools: Python • Pandas • NumPy • SQL • Power BI/Tableau
Dataset: Google Play Store Apps & User Reviews

# Google Play Store Intelligence Dashboard

## Data Understanding

### Project Overview

This project analyzes the Google Play Store dataset to uncover business insights, identify app market trends, and recommend opportunities for developers using data analytics.

The project follows a complete data analytics workflow:

1. Data Understanding
2. Data Cleaning
3. Feature Engineering
4. Exploratory Data Analysis (EDA)
5. SQL Business Analysis
6. Dashboard Development
7. Business Recommendations

---

# Business Problem

The Google Play Store contains thousands of applications across multiple categories. Developers often struggle to identify:

- Which app categories perform the best?
- What factors contribute to app success?
- How do ratings, reviews, pricing, and installs relate?
- Which market segments offer the greatest opportunity?

The objective of this project is to answer these questions through data analytics and visualization.

# Objectives

This notebook focuses on understanding the datasets before any cleaning or transformation.

The objectives are:

- Understand dataset structure
- Explore columns and data types
- Identify missing values
- Identify duplicate records
- Detect invalid values
- Document data quality issues

# Import Required Libraries

In [4]:
import pandas as pd
import numpy as np

pd.set_option("display.max_columns", None)

# Load the Datasets

In [5]:
apps = pd.read_csv("../data/raw/googleplaystore.csv")
reviews = pd.read_csv("../data/raw/googleplaystore_user_reviews.csv")

# Preview the Data

Let's inspect the first few rows of each dataset to understand their structure.

In [6]:
apps.head()

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
0,Photo Editor & Candy Camera & Grid & ScrapBook,ART_AND_DESIGN,4.1,159,19M,"10,000+",Free,0,Everyone,Art & Design,"January 7, 2018",1.0.0,4.0.3 and up
1,Coloring book moana,ART_AND_DESIGN,3.9,967,14M,"500,000+",Free,0,Everyone,Art & Design;Pretend Play,"January 15, 2018",2.0.0,4.0.3 and up
2,"U Launcher Lite – FREE Live Cool Themes, Hide ...",ART_AND_DESIGN,4.7,87510,8.7M,"5,000,000+",Free,0,Everyone,Art & Design,"August 1, 2018",1.2.4,4.0.3 and up
3,Sketch - Draw & Paint,ART_AND_DESIGN,4.5,215644,25M,"50,000,000+",Free,0,Teen,Art & Design,"June 8, 2018",Varies with device,4.2 and up
4,Pixel Draw - Number Art Coloring Book,ART_AND_DESIGN,4.3,967,2.8M,"100,000+",Free,0,Everyone,Art & Design;Creativity,"June 20, 2018",1.1,4.4 and up


In [7]:
reviews.head()

,App,Translated_Review,Sentiment,Sentiment_Polarity,Sentiment_Subjectivity
0,10 Best Foods for You,I like eat delicious food. That's I'm cooking ...,Positive,1.00,0.533333
1,10 Best Foods for You,This help eating healthy exercise regular basis,Positive,0.25,0.288462
2,10 Best Foods for You,NaN,NaN,NaN,NaN
3,10 Best Foods for You,Works great especially going grocery store,Positive,0.40,0.875000
4,10 Best Foods for You,Best idea us,Positive,1.00,0.300000


#  Dataset Dimensions

Understanding the number of rows and columns provides an overview of the size of each dataset.

In [8]:
print(f"Apps Dataset Shape: {apps.shape}")
print(f"Reviews Dataset Shape: {reviews.shape}")

Apps Dataset Shape: (10841, 13)
Reviews Dataset Shape: (64295, 5)


# Column Names

Let's inspect the available features in each dataset.

In [9]:
print(apps.columns.tolist())

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


In [10]:
print(reviews.columns.tolist())

['App', 'Translated_Review', 'Sentiment', 'Sentiment_Polarity', 'Sentiment_Subjectivity']


#  Data Types

Checking data types helps identify columns that require conversion before analysis.

In [11]:
apps.dtypes

App                   str
Category              str
Rating            float64
Reviews               str
Size                  str
Installs              str
Type                  str
Price                 str
Content Rating        str
Genres                str
Last Updated          str
Current Ver           str
Android Ver           str
dtype: object

In [12]:
reviews.dtypes


App                           str
Translated_Review             str
Sentiment                     str
Sentiment_Polarity        float64
Sentiment_Subjectivity    float64
dtype: object

#  Missing Values Analysis

Missing values can affect analysis and machine learning models. We identify them before deciding how to handle them.

In [13]:
apps.isnull().sum().sort_values(ascending=False)

Rating            1474
Current Ver          8
Android Ver          3
Content Rating       1
Type                 1
Size                 0
Reviews              0
Category             0
App                  0
Price                0
Installs             0
Last Updated         0
Genres               0
dtype: int64

In [14]:
reviews.isnull().sum().sort_values(ascending=False)

Translated_Review         26868
Sentiment_Polarity        26863
Sentiment                 26863
Sentiment_Subjectivity    26863
App                           0
dtype: int64

#  Duplicate Records

Duplicate rows may introduce bias in the analysis. We identify them before cleaning.

In [15]:
print("Apps Dataset Duplicates:", apps.duplicated().sum())
print("Reviews Dataset Duplicates:", reviews.duplicated().sum())

Apps Dataset Duplicates: 483
Reviews Dataset Duplicates: 33616


# Summary Statistics

Generate descriptive statistics to understand the distribution of the data.

In [16]:
apps.describe(include="all")

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
count,10841,10841,9367.000000,10841,10841,10841,10840,10841,10840,10841,10841,10833,10838
unique,9660,34,NaN,6002,462,22,3,93,6,120,1378,2832,33
top,ROBLOX,FAMILY,NaN,0,Varies with device,"1,000,000+",Free,0,Everyone,Tools,"August 3, 2018",Varies with device,4.1 and up
freq,9,1972,NaN,596,1695,1579,10039,10040,8714,842,326,1459,2451
mean,NaN,NaN,4.193338,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
std,NaN,NaN,0.537431,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
min,NaN,NaN,1.000000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
25%,NaN,NaN,4.000000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
50%,NaN,NaN,4.300000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
75%,NaN,NaN,4.500000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


# Initial Data Quality Observations

Based on the exploration, several issues are immediately visible:

- Missing values exist in the Rating column.
- Reviews, Installs, and Price are stored as text instead of numeric values.
- Duplicate records are present.
- Some rows contain corrupted values.
- The Size column contains inconsistent units (MB, KB, "Varies with device").

These issues will be addressed in the next notebook.

#  Data Quality Report

| Issue | Observation |
|--------|-------------|
| Missing Ratings | 1474 missing values |
| Duplicate Apps | 483 duplicate rows |
| Duplicate Reviews | 33,616 duplicate rows |
| Incorrect Data Types | Reviews, Installs, Price |
| Corrupted Record | Rating value greater than 5 |
| Inconsistent Size | Mixed units and "Varies with device" |

#  Key Findings

From the initial exploration, we identified several data quality issues that must be resolved before analysis.

The next notebook will focus on:

- Removing corrupted records
- Removing duplicate rows
- Converting columns to appropriate data types
- Preparing a clean dataset for feature engineering

# Next Step

Proceed to:

## 02_Data_Cleaning.ipynb

where the dataset will be cleaned and transformed into an analysis-ready format.